lib/src/serializable.dart

```diff
@@ -0,0 +1,224 @@
+    }
+
+    // Check here that the position args in data match the expected params of the method being called
+    if (apiMethodMirror.parameters.length == data.length) {
+      for (var i = 0; i < apiMethodMirror.parameters.length; i++) {
+        var param = apiMethodMirror.parameters[i];
+
+        if (data[i] is Map && param.type.reflectedType != Map) {
+          ClassMirror paramClassMirror = reflectClass(param.type.reflectedType);
+
+          // Paramter type must implement fromJson name constructor that takes a Map
+          try {
+            var instance = paramClassMirror
+                .newInstance(new Symbol('fromJson'), [data[i]]).reflectee;
+            data[i] = instance;
+          } on NoSuchMethodError {
+            _logger.warning(
+                '${paramClassMirror.simpleName.toString()} does not implement fromJson named constructor');
+            return;
+          }
+        }
+      }
+
+      try {
+        apiMirror.invoke(new Symbol(method), data);
+      } catch (e) {
+        _logger.severe(
+            'Unable to call $method on ${module.serializableKey} w_module, ${e.toString()}');
+      }
+    } else {
+      _logger.warning(
+          'Unable to call api method $method in ${module.serializableKey} w_module, mismatched params');
+    }
+  }
+
+  void _registerForAllEvents(_ModuleRegistration registration) {
+    if (registration.module.events != null) {
+      for (var event in registration.module.events.allEvents) {
+        if (event is SerializableEvent) {
+          registration.manageStreamSubscription(event.listen((payload) =>
+              _sendEvent(registration.module, event.eventKey, payload)));
+        }
+      }
+    } else {
+      _logger.info(
+          'Events not defined for ${registration.module.serializableKey} module');
+    }
+  }
+
+  void _registerForLifecycleEvents(_ModuleRegistration registration) {
+    registration
+        .manageStreamSubscription(registration.module.willLoad.listen((_) {
+      _registerForAllEvents(registration);
+      _sendEvent(registration.module, 'willLoad', null);
+    }));
+    registration.manageStreamSubscription(registration.module.didLoad
+        .listen((_) => _sendEvent(registration.module, 'didLoad', null)));
+    registration.manageStreamSubscription(registration.module.willUnload
+        .listen((_) => _sendEvent(registration.module, 'willUnload', null)));
+    registration
+        .manageStreamSubscription(registration.module.didUnload.listen((_) {
+      _sendEvent(registration.module, 'didUnload', null);
+      deregisterModule(registration.module);
+    }));
+  }
+
+  void _sendEvent(SerializableModule module, String eventKey, Object data) {
+    Map event = <String, dynamic>{};
+    event['module'] = module.serializableKey;
+    event['event'] = eventKey;
+
+    if (data is JsonSerializable) {
+      data = (data as JsonSerializable).toJson();
+    }
+
+    event['data'] = data;
+
+    _bridge?.broadcastSerializedEvent(event);
```

Comment:
Do we want to know if the `_bridge` is null at the time that we attempt to send an event?  If so, we may consider adding some logging around that.